### ISSUE TITLE
REJECTED_CANONICAL

### ISSUE THREAD

**Issue by csaezl** [Norconex/collector-http#112]
Since I use 2.2.0-SNAPSHOT I get a lot of ```REJECTED_CANONICAL``` events. If they reject the document in favor of the canonical, that's right. But I have a case that is not working properly for me. 

From the log:
```
MC (crawler): 2015-05-24 19:25:05 INFO -          DOCUMENT_FETCHED: http://europa.eu/youth/eu/article/57/1281_de
MC (crawler): 2015-05-24 19:25:05 WARN - Not able to obtain robots.txt at: /youth/eu/article/57/1281_de/robots.txt
org.apache.http.client.ClientProtocolException
...
Caused by: org.apache.http.ProtocolException: Target host is not specified
...
MC (crawler): 2015-05-24 19:25:05 INFO -           REJECTED_FILTER: /youth/eu/article/57/1281_de
MC (crawler): 2015-05-24 19:25:05 INFO -        REJECTED_CANONICAL: http://europa.eu/youth/eu/article/57/1281_de
MC (crawler): 2015-05-24 19:25:05 INFO -           REJECTED_IMPORT: http://europa.eu/youth/eu/article/57/1281_de

```
The document URL is ```http://europa.eu/youth/eu/article/57/1281_de``` and from its source: ```<link rel="canonical" href="/youth/eu/article/57/1281_de" />```

The document is ```REJECTED CANONICAL``` and there is not registered document for the canonical URL.

Is there anything I can do?

**Comment by essiembre** [comment]
I suspect a regression issue introduced in the latest snapshot with resolving relative URLs.  Investigating...

**Comment by essiembre** [comment]
Turned out to be a different issue.  Canonical URLs found in HTML ``<head>`` section were assumed to be absolute.  I changed the code so relative ones are now made absolute. I also made a fix so that the "Link" in your HTTP Headers is parsed properly (it was not).  That one will be detected before the one in the HTML content is.  These fixes are in a new [snapshot release](http://www.norconex.com/collectors/collector-http/download) I just made.
